### PR TITLE
Remove RenderSprites animation keys.

### PR DESCRIPTION
**Pull request by pchote**
The code that previously required these was removed ages ago, and they now just cause problems (duplicate key exceptions when adding multiple traits).  Conflicts with several existing prs.

### CODE REVIEW

**Comment by Mailaender** [comment]
The underscore notation always felt alien and string comparisons probably were not cheap so I am glad this is gone now. :+1:

File: OpenRA.Mods.D2k/Traits/Render/WithDeliveryOverlay.cs [hunk 1]
```diff
@@ -46,21 +48,23 @@ public WithDeliveryOverlay(Actor self, WithDeliveryOverlayInfo info)
 			var rs = self.Trait<RenderSprites>();
 			var body = self.Trait<IBodyOrientation>();
 
-			buildComplete = !self.HasTrait<Building>(); // always render instantly for units
+			// always render instantly for units
+			buildComplete = !self.HasTrait<Building>();
```
Comment (Mailaender) [inline OpenRA.Mods.D2k/Traits/Render/WithDeliveryOverlay.cs]: [SP2000] Invalid spacing at the end of the line.

**Comment by Mailaender** [comment]
Needs a rebase. Let's merge this first. The rebase conflicts for the other pull requests will be probably be easy to solve.

**Comment by Mailaender** [comment]
Don't forget about the recent addition https://github.com/OpenRA/OpenRA/pull/7862.

**Comment by pchote** [comment]
Rebased.

**Comment by Mailaender** [comment]
Needs a rebase again.

**Comment by pchote** [comment]
Rebased again.

File: OpenRA.Mods.Common/Traits/Render/WithRotor.cs [hunk 2]
```diff
@@ -28,9 +28,6 @@ public class WithRotorInfo : ITraitInfo, IRenderActorPreviewSpritesInfo, Require
 		[Desc("Position relative to body")]
 		public readonly WVec Offset = WVec.Zero;
 
-		[Desc("Change this when using this trait multiple times on the same actor.")]
-		public readonly string Id = "rotor";
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Traits/Render/WithRotor.cs]: Do we need/want an upgrade rule for this?

Comment (Mailaender) [inline OpenRA.Mods.Common/Traits/Render/WithRotor.cs]: We don't need it as the game won't error if these stay in the rules, but I would like to see the now junk rules removed from https://github.com/pchote/OpenRA/blob/animation-keys/mods/ra/rules/aircraft.yaml#L213 etc.

Comment (penev92) [inline OpenRA.Mods.Common/Traits/Render/WithRotor.cs]: We should clean the YAML in such cases. If we start leaving stuff like this soon half of our YAML files will be unused junk values.

**Comment by pchote** [comment]
Updated.

**Comment by abcdefg30** [comment]
Looks good to me and didn't spot anything ingame. :+1: / :white_check_mark:

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/031395c4e067457099f8aa87a0e5b7cb63f9bf02...779f7af64ce0ff35c8c57b01e18e226c693003f2)

**Pull request merged**